milestone_1/crop_yield/crop_yield.md

```diff
@@ -0,0 +1,297 @@
+    x, y = MODIS_GRID(lon, lat)
+
+    # Calculate horizontal and vertical grid indices (h and v)
+    h = (EARTH_WIDTH * 0.5 + x) / TILE_WIDTH
+    v = -(EARTH_WIDTH * 0.25 + y - (VERTICAL_TILES - 0) * TILE_HEIGHT) / TILE_HEIGHT
+
+    # Return the grid indices as integers
+    return int(h), int(v)
+
+
+# Example usage
+x, y = lat_lon_to_modis(51.7422767, 5.9325433)
+# 5.0115545, 51.8170254
+# 6.8113423, 51.9661496
+# 5.9325433, 51.7422767
+print(x)
+print(y)
+```
+
+```python
+"""""" """
+Product must be same name from the list at https://e4ftl01.cr.usgs.gov/MOLT/
+""" ""
+# Define the parameters
+product = "MOD09A1.061"  # e.g., 'MOD09A1'
+tiles = "h18v03"  # e.g., 'h17v03,h18v03'
+# Gelderland falls in 18th Horizontally and third vertically.
+if not os.path.exists("NIR"):
+    os.makedirs("NIR")
+output_directory = "./NIR/"  # Output Directory
+day = "2023-08-01"  # Starting Day
+endDate = "2023-08-31"  # Ending Day
+
+# Initialize the downloader
+modis_downloader = downmodis.downModis(
+    product=product,
+    tiles=tiles,
+    user="hellium",  # Change to your username
+    password="Siksik2010##",  # Change to your password
+    destinationFolder=output_directory,
+    today=day,
+    enddate=endDate,
+    jpg=True,
+)
+
+# Download the data
+modis_downloader.connect()
+modis_downloader.downloadsAllDay()
+```
+
+```python
+# Get List of Downloaded Files
+files = glob.glob(os.path.join(output_directory, "MOD09A1*.hdf"))
+# Initialize Dictionary to Store NDVI Values And List to Store Temporal Values
+print(output_directory)
+tempList = []
+NDVI_DICT = {}
+
+for file in files:
+    # Get Day Value From Filename
+    # Given filename for example is MOD09A1.A2023217.h18v03.061.2023226041025.hdf
+    # We want to get the day value of the file
+    # So we split the filename on the period and then on the underscore
+    # Then we take the second item in the list which is the day value
+    # We then convert it to an integer
+    # We convert it using datetime using year and day
+    # Subtract 1 from the day value to get the correct date
+    day_value = int(file.split("/")[-1].split(".")[1][5:])
+    year = 2023
+    date = datetime.datetime(year, 1, 1) + datetime.timedelta(day_value - 1)
+    # Read hdf file and get the red and nir values
+    hdf = gdal.Open(file, gdal.GA_ReadOnly)
+    # Get the dictionary of hdf metadata
+    hdfDict = hdf.GetMetadata("SUBDATASETS")
+    # Get the list of hdf metadata keys that contain the name of the hdf file
+    hdfList = [hdfDict[k] for k in hdfDict.keys() if "_NAME" in k]
+    # Open each hdf file and get the red and nir values
+    for n in hdfList:
+        tempList.append(gdal.Open(n))
+    # RED_VALUES is the first band sur_refl_b01 (620-670 nm)
+    RED_VALUES = tempList[0].GetRasterBand(1).ReadAsArray().astype(np.int16)
+    # NIR_VALUES is the second band sur_refl_b02 (841-876 nm)
+    NIR_VALUES = tempList[1].GetRasterBand(1).ReadAsArray().astype(np.int16)
+    # Calculate NDVI Values
```

Comment:
after getting red and nir values, clip for the geography of choice and only then compute NDVI